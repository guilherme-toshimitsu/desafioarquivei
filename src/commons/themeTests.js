import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
