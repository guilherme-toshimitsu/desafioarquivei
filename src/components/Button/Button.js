import { memo } from "react";
import styled, { css } from "styled-components";

const StyledButton = styled.button.attrs((props) => ({
  disabled: props.disabled || false,
}))`
  width: 100px;
  min-height: 40px;
  border: 0px;
  border-radius: 4px;
  margin: 0 12px 0 12px;
  color: white;
  font-size: 16px;
  outline: none;
  background-color: ${({ theme }) => theme.colors.main}
    ${({ disabled }) =>
      disabled &&
      css`
        height: 60px;
        background-color: ${({ theme }) => theme.colors.disabled};
      `};
`;

export default memo(StyledButton);
