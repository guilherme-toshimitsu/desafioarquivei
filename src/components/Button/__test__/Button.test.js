import React from "react";
import Button from "../";
import { renderWithTheme } from "@commons/themeTests";

describe("<Button />", () => {
  it("creates a Button component", () => {
    const wrapper = renderWithTheme(<Button />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
