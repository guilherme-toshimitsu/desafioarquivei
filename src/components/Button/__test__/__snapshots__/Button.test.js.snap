// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> creates a Button component 1`] = `
<DocumentFragment>
  <button
    class="Button__StyledButton-pzevnv-0 iLzPQp"
  />
</DocumentFragment>
`;
